### ISSUE TITLE
Enable HonoConnection to create a sender link on the anonymous relay

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#1541]
Currently, `HonoConnection.createSender()` doesn't allow the link target address to be `null`.

That is however a valid use case since a `null` address is used to establish a sender link to the anonymous relay (see [here](http://docs.oasis-open.org/amqp/anonterm/v1.0/anonterm-v1.0.html) or the [ProtonConnection](https://github.com/vert-x3/vertx-proton/blob/c02a1a4bce17daf8626c9eb11afd30857ffdbf02/src/main/java/io/vertx/proton/ProtonConnection.java#L77) class).

And that is especially useful for creating a telemetry/event sender link to the Hono AMQP adapter, which requires such a link on the anonymous relay.

**Comment by calohmn** [comment]
We have already used the `HonoConnection.createSender()` method for creating anonymous relay links in connection with the Qpid dispatch router. For that we have used an empty target address (instead of `null`). Qpid seems to support that.
Nonetheless we should support a `null` address since that is specified in the anonterm spec and the ProtonConnection class. The check in the AMQP adapter class is for a `null` value as well (that's why `HonoConnection.createSender()` can't be used with the AMQP adapter as target as of now).

**Comment by calohmn** [comment]
The empty target address already seems to be treated in a special way by Proton, since `link.getRemoteTarget().getAddress()` is `null` in that case.
Therefore, when initially adding support for the empty string, we needed to adapt the `isLinkEstablished` method:
https://github.com/eclipse/hono/blob/20d05db99c6f66ef5b0a2c2750bddd61ea773439/core/src/main/java/org/eclipse/hono/util/HonoProtonHelper.java#L145-L150
(see https://github.com/eclipse/hono/commit/36e6748816b14731448ccbbdad209f92c3c3f703#diff-942667be6d4a8d180ac40f1520d28858R150).

I've created PR #1542 now, still supporting the empty string address, but not labelling it as the anonymous relay link target.
